### PR TITLE
Add a test case to increase the nodecount of a running instance

### PR DESCRIPTION
**Pull request by kevinearls**
Signed-off-by: Kevin Earls <kearls@redhat.com>

### CODE REVIEW

**Comment by kevinearls** [comment]
@objectiser @jpkrohling  @pavolloffay Please review

File: test/e2e/self_provisioned_elasticsearch_test.go [hunk 1]
```diff
@@ -76,18 +80,56 @@ func (suite *SelfProvisionedTestSuite) AfterTest(suiteName, testName string) {
...
+	err := fw.Client.Create(goctx.TODO(), exampleJaeger, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
+	require.NoError(t, err, "Error deploying example Jaeger")
+	defer undeployJaegerInstance(exampleJaeger)
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, jaegerInstanceName+"-collector", 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for collector deployment")
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, jaegerInstanceName+"-query", 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for query deployment")
+
```
Comment (objectiser) [inline test/e2e/self_provisioned_elasticsearch_test.go]: Might be worth doing a run of the smoke test here, to confirm all running ok, before the update.

File: test/e2e/self_provisioned_elasticsearch_test.go [hunk 2]
```diff
@@ -76,18 +80,56 @@ func (suite *SelfProvisionedTestSuite) AfterTest(suiteName, testName string) {
...
+	updateNodeCount := 2
+	changeNodeCount(jaegerInstanceName, updateNodeCount)
+	updatedJaegerInstance := getJaegerInstance(jaegerInstanceName, namespace)
+	require.EqualValues(t, updateNodeCount, updatedJaegerInstance.Spec.Storage.Elasticsearch.NodeCount)
+
+	// Make sure there are 2 ES deployments. Note: The deployment name is based on  "elasticsearch-cdm-" + namespace + jaegerInstanceName + "-1"
+	// with dashes in the namespace and  jaegerInstanceName removed
+	for i := 1; i <= updateNodeCount; i++ {
```
Comment (objectiser) [inline test/e2e/self_provisioned_elasticsearch_test.go]: Although testing an update of the ES config is useful - we primarily need to test an update on the Jaeger components themselves. So possibly also increase query and collector replicas?

**Comment by kevinearls** [comment]
Replaced by #824 because of problems I had with dependency checksums after the update to operator sdk 0.12 and golang 1.13

**Pull request closed without merging**